Sources/Purchasing/StoreKitAbstractions/StoreProductDiscount.swift

```diff
@@ -114,6 +114,21 @@ public final class StoreProductDiscount: NSObject, StoreProductDiscountType {
     }
 
 }
+extension StoreProductDiscount {
+    // swiftlint:disable:next missing_docs
+    @_spi(Internal) public func promotionalOffer(withSignedDataIdentifier identifier: String,
```

Comment:
Hmm I'm wondering if it would make sense to provide a public initializer for `StoreProductDiscount` so people can write more tests... Or maybe the more Swifty approach would be to create a protocol and use that instead of the class, so it can be implemented as needed... Still, I think this is the best approach for now 👍

Reply:
Yeah, that's a good point. However, we do already expose the [`TestStoreProductDiscount`](https://github.com/RevenueCat/purchases-ios/blob/ac6da565f07d6048ddde257c21342add7cd00e5c/Sources/Purchasing/StoreKitAbstractions/Test%20Data/TestStoreProductDiscount.swift), which is (I think) exactly that, and can be mapped into a `StoreProductDiscount` object.
However, we do not have the same for `PromotionalOffer`, which I needed to create from the app. And because the `init` is `internal`, this way minimizes the APIs that need to be made public for my used case (which is creating the promotional offer associated to an existing `StoreProductDiscount` object)